html2text/__init__.py

```diff
@@ -537,8 +537,7 @@ def link_url(self: HTML2Text, link: str, title: str = "") -> None:
                             self.o("][" + str(a_props.count) + "]")
 
         if tag == "img" and start and not self.ignore_images:
-            if "src" in attrs:
-                assert attrs["src"] is not None
+            if "src" in attrs and attrs["src"] is not None:
```

Comment:
Maybe this could be simplified by 

```suggestion
            if attrs.get("src"):
```

Or walrus:

```suggestion
            if img_src := attrs.get("src"):
```

The same approach could be applied to the rest of new code introduced in this pull request.

Reply:
`if attrs.get("src"):` changes the behaviour when the value is the empty string.  `if attrs.get("src") is not None:` fixes that, but neither of them lets mypy narrow the type -- within that block, `attrs["src"]` is still `Optional[str]`.

Reply:
Makes sense.
Thanks for the clarification.